sdk/python/lib/pulumi/resource.py

```diff
@@ -898,7 +906,7 @@ def create_urn(
         if project is None:
             project = get_project()
 
-        parent_prefix = "urn:pulumi:" + stack + "::" + project + "::"
-
-    return Output.all(parent_prefix, type_, name).apply(
+        parent_prefix = Output.from_input("urn:pulumi:" + stack + "::" + project + "::")
+    all_args = [parent_prefix, type_, name]
+    return Output.all(all_args).apply(
```

Comment:
Was this code just broken before hand?  Is the splitting into two lines strctly necessary for `mypy` or was this just about adding the `[`/`]`?

Reply:
For mypy. I wind up having to `type ignore` due to an invariant list anyways (update after fixing tests). 

The original error:

`lib/pulumi/resource.py:913: error: Argument 1 to "all" of "Output" has incompatible type "Output[str]"; expected "List[Union[<nothing>, Awaitable[<nothing>], Output[<nothing>]]]"
lib/pulumi/resource.py:913: error: Argument 2 to "all" of "Output" has incompatible type "Union[str, Awaitable[str], Output[str]]"; expected "List[Union[<nothing>, Awaitable[<nothing>], Output[<nothing>]]]"
lib/pulumi/resource.py:913: error: Argument 3 to "all" of "Output" has incompatible type "Union[str, Awaitable[str], Output[str]]"; expected "List[Union[<nothing>, Awaitable[<nothing>], Output[<nothing>]]]"
lib/pulumi/resource.py:913: error: Unsupported left operand type for + (<nothing>)
`